clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -6037,6 +6037,9 @@ void CodeGenModule::EmitTopLevelDecl(Decl *D) {
     // File-scope asm is ignored during device-side OpenMP compilation.
     if (LangOpts.OpenMPIsDevice)
       break;
+    // File-scope asm is ignored during device-side SYCL compilation.
+    if (LangOpts.SYCLIsDevice)
+      break;
```

Comment:
I suggest we commit CodeGenModule.cpp change with a new test to llorg directly and revert https://github.com/intel/llvm/commit/2bc91937a7ea1a2ef4ec9f510d7bf34d926bf898 after these changes are pulled from llorg.

Reply:
I uploaded changes for review https://reviews.llvm.org/D96538.
But can we commit changes here as well to not wait for the pulldown?

Reply:
> I uploaded changes for review https://reviews.llvm.org/D96538.

Thank you!

> But can we commit changes here as well to not wait for the pulldown?

Yes, although I think wait for pulldown is better strategy if getting fix is not urgent.
1. You save your time on doing commit twice.
2. If patches committed to llorg and sycl branch are different (e.g. address code review comments), we will have to resolve merge conflicts. There are such merge conflicts with pulldown approach.

If you are okay to add this (relatively) small overhead to get the patch early in the sycl branch, you can commit to both branches.